### ISSUE TITLE
False positive when declared class member is referenced as a subclass instance member

### ISSUE THREAD

**Issue by jolaf** [PyCQA/pylint#3143]
### Steps to reproduce
```python3
class A:
    x: str

    @classmethod
    def init(cls) -> None:
        cls.x = "OK"

class B(A):
    pass

A.init()
print(B().x)
```
```
$ python3 Test.py 
OK
$ pylint3 Test.py
************* Module Test
Test.py:12: [E1101(no-member), ] Instance of 'B' has no 'x' member
```
### Current behavior
`E1101` is emitted
### Expected behavior
No warning emitted
### pylint --version output
```
pylint 2.4.1
astroid 2.3.0
Python 3.6.8 (default, Aug 20 2019, 17:12:48) 
[GCC 8.3.0]
```

**Comment by jolaf** [comment]
Still actual for v2.4.2

**Comment by jolaf** [comment]
Marked for next minor release, but still unfixed in 2.4.4. :(

**Comment by PCManticore** [comment]
@jolaf Feel free to work on a patch if you want this fixed that much.

**Comment by jolaf** [comment]
Somehow looks fixed on v2.5.2 on Windows, but not on Ubuntu.

**Comment by DanielNoord** [comment]
@Pierre-Sassoulas I think this might have been fixed. At least, on my machine it is not throwing the message anymore.
